doc/building_fpga_bitstreams.md

```diff
@@ -0,0 +1,100 @@
+# Building FPGA Bitstreams for OT-SCA
+
+The idea of this tutorial is to document how to build FPGA bitstreams that can
+be used for performing side-channel analysis on the ChipWhisperer boards using
+ot-sca.
+
+## Prerequisites
+
+Bitstreams are created inside the [OpenTitan](https://github.com/lowRISC/OpenTitan)
+repository. Make sure to follow the getting started
+[guide](https://opentitan.org/book/doc/getting_started/index.html) before attempting
+to build bitstreams. A Xilinx Vivado installation with a valid license is needed.
+
+Then, clone the OpenTitan repository and select a suitable commit. Make sure that
+the selected commit is compatible with the penetration testing
+[framework](https://github.com/lowRISC/opentitan/tree/master/sw/device/tests/penetrationtests).
```

Comment:
What does it mean to be compatible with the penetration testing framework?

Reply:
Tricky question and I do not know how to describe this in the document. If somebody uses the same git commit for building the FPGA and the framework, everything should work. However, it is getting complicated if somebody want to have a specific HW version.

Example:
If you want to build the FPGA bitstream for the Z1 HW state, you can use the earlgrey_es_sival OpenTitan commit. However, in this commit the new pen. testing framework did not exist and when using the pen. testing framework from the master commit of OpenTitan, the SW driver is not compatible with Z1.
When using the A1 HW state (earlgrey_1.0.0 branch) you need to fetch the pen. testing framework from the master branch as in the earlgrey_1.0.0 the pen. testing framework is not up to date.

I think this is a bigger problem that we do not have a versioning of HW and SW in the project.

Reply:
Understood.